projects/slap_manipulation/src/slap_manipulation/agents/language_ovmm_agent.py

```diff
@@ -0,0 +1,233 @@
+import logging
+from enum import Enum
+from glob import glob
+from typing import Any, Dict, List, Tuple
+
+import pandas as pd
+
+from home_robot.agent.ovmm_agent.pick_and_place_agent import (
+    PickAndPlaceAgent,
+    SimpleTaskState,
+)
+from home_robot.core.interfaces import Action, DiscreteNavigationAction, Observations
+
+
+class GeneralTaskState(Enum):
+    NOT_STARTED = 0
+    PREPPING = 1
+    DOING_TASK = 2
+    IDLE = 3
+    STOP = 4
+
+
+def get_task_plans_from_gt(
+    index, datafile="./datasets/BringXFromYSurfaceToHuman.json", root="./datasets/"
+):
+    """Reads the dataset files and return a list of task plans"""
+    if datafile == "all":
+        files = glob(root + "*.json")
+        dflist = []
+        for file in files:
+            dflist.append(pd.read_json(file))
+        df = pd.concat(dflist)
+    else:
+        df = pd.read_json(datafile)
+    assert index < len(df), f"Index {index} is out of range"
+    steps_list = df.iloc[index]["steps"]
+    # steps_df = pd.DataFrame.from_records(steps_list)
+    code = get_codelist(steps_list)
+    return code
+
+
+def get_codelist(steps_list):
+    codelist = []
+    for step in steps_list:
+        codelist += [
+            f"self.{step['verb']}('{step['noun']}', motion_profile={step['adverb']}, obs=obs)"
```

Comment:
these are the specific commands to execute right?

Reply:
@vidhiJain